src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -82,15 +84,34 @@ private class ProgressChangedData
         /// Initializes a new instance of the <see cref="ProgressDialog"/> class.
         /// </summary>
         public ProgressDialog()
-            : this(null)
+            : this(null, System.Threading.CancellationToken.None)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProgressDialog"/> class, adding it to the specified container.
+        /// </summary>        
+        /// <param name="token"></param>
+        public ProgressDialog(System.Threading.CancellationToken token)
+            : this(null, token)
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ProgressDialog"/> class, adding it to the specified container.
         /// </summary>
-        /// <param name="container">The <see cref="IContainer"/> to which the component should be added.</param>
+        /// <param name="container">The <see cref="IContainer"/> to which the component should be added.</param>        
         public ProgressDialog(IContainer container)
+            : this(container, System.Threading.CancellationToken.None)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProgressDialog"/> class, adding it to the specified container.
+        /// </summary>
+        /// <param name="container">The <see cref="IContainer"/> to which the component should be added.</param>
+        /// <param name="token"></param>
+        public ProgressDialog(IContainer container, System.Threading.CancellationToken token)
```

Comment:
The `ProgressDialog` has been designed to be reused more than once to report progress, so the user could create a single instance of the `ProgressDialog` and then call `Show` or `ShowDialog` for different jobs (in sequence)... Different jobs could have different `CancellationToken`s, so I'm thinking it would best to receive the `CancelationToken` via `Show` and `ShowDialog` rather than via constructor.

Reply:
yes, that can be moved to the show* methods

Reply:
The `DoWork` receives a reference to function that will not be executed until you call `ShowDialog` later, so I'm not sure why you think we need to know the exact `CancellationToken` at the time we register the `DoWork` event :

```csharp
var dlg = new Ookii.Dialogs.Wpf.ProgressDialog();
                
dlg.DoWork += (s, e) =>
{
   // This won't be run until ShowDialog is called
   MyJob(dlg, e.CancellationToken); // Linked from the original token provided by the user (if any)
};

dlg.ShowDialog(cancellationToken); // Provided by the user (optional)
```